tests/gtest/avifincrtest_helpers.cc

```diff
@@ -332,6 +331,13 @@ avifResult DecodeIncrementally(const avifRWData& encoded_avif,
     data.available.size = std::min(data.available.size + step, data.full_size);
     parse_result = avifDecoderParse(decoder);
   }
+  if (data.available.size == data.full_size) {
+    // Can happen if the data is in 'idat', or if some metadata is at the end of
+    // the file. But ideally this should be avoided.
+    printf(
+        "WARNING: had to provide the whole file for avifDecoderParse() to "
+        "succeed\n");
+  }
```

Comment:
I would check this only if an arg given to `DecodeIncrementally()` is true and return an error or abort here. This way it can be silent for fuzzed encoded bytes but strongly enabled for encoded bytes generated by libavif.